lxd/container_new.go

```diff
@@ -174,36 +215,288 @@ func (c *lxdContainer) isPrivileged() bool {
 	return false
 }
 
+func (c *lxdContainer) IsRunning() bool {
+	return c.c.Running()
+}
+
 func (c *lxdContainer) Shutdown(timeout time.Duration) error {
-	return c.c.Shutdown(timeout)
+	if err := c.c.Shutdown(timeout); err != nil {
+
+		// TODO: Not sure this line is right.
```

Comment:
If shutdown fails, the container is likely still running, so stopping the storage will likely fail too, but I guess it doesn't hurt trying to do it anyway, just in case.